File: university-service/src/main/java/ua/com/foxminded/courseproject/repository/TeacherRepository.java
package ua.com.foxminded.courseproject.repository;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.repository.CrudRepository;
import org.springframework.stereotype.Repository;
import ua.com.foxminded.courseproject.entity.Teacher;

import java.time.LocalDate;
import java.util.Optional;
import java.util.UUID;

@Repository
public interface TeacherRepository extends CrudRepository<Teacher, UUID> {
    Page<Teacher> findAll(Pageable pageable);

    Optional<Teacher> findById(UUID id);

    Teacher save(Teacher teacher);

    void delete(Teacher teacher);

    Boolean existsTeacherByFirstNameAndLastNameAndBirthDay(String firstName, String lastName, LocalDate birthDay);
}

File: university-service/src/main/java/ua/com/foxminded/courseproject/repository/TeacherRepository.kt
package ua.com.foxminded.courseproject.repository

import org.springframework.data.repository.reactive.ReactiveSortingRepository
import reactor.core.publisher.Flux
import reactor.core.publisher.Mono
import ua.com.foxminded.courseproject.entity.Teacher
import java.time.LocalDate
import java.util.*

interface TeacherRepository : ReactiveSortingRepository<Teacher, UUID> {
    override fun findAll(): Flux<Teacher>
    override fun findById(id: UUID): Mono<Teacher>
    fun save(teacher: Teacher): Mono<Teacher>
    override fun delete(teacher: Teacher): Mono<Void>
    fun existsTeacherByFirstNameAndLastNameAndBirthDay(
        firstName: String,
        lastName: String,
        birthDay: LocalDate
    ): Mono<Boolean>
}